### PR TITLE
Include Java example code excerpts

### PR DESCRIPTION
**Pull request by zeitlinger**

### CODE REVIEW

File: .gitmodules
Comment (zeitlinger) [inline .gitmodules:33]: this should be replaced once https://github.com/open-telemetry/opentelemetry-java-examples/pull/422 is merged

**Comment by zeitlinger** [comment]
@svrnm can you help with the build error?

**Comment by svrnm** [comment]
> @svrnm can you help with the build error?

@chalin knows best how the code excerpts work, let's wait for him to be back and take a look!

**Review by tiffany76 — APPROVED** [review]

**Comment by svrnm** [comment]
@chalin can you take a look? Or should we currently not introduce more pages that use the code excerpt?

**Comment by chalin** [comment]
Thanks for the ping and your patience. Great work here @zeitlinger!

Could you rebase and resolve the conflicts and I'll take a look. I'm OOO next week, but try to work on this PR tomorrow if you can get the conflicts resolved by then.

> @chalin can you take a look? Or should we currently not introduce more pages that use the code excerpt?

I think that it is worth bring in this PR.

Oh that build error was tricky to resolve. I can't recall how I worked through it the last time, but I'll find out when I next work on this PR.

**Comment by zeitlinger** [comment]
@chalin merge conflict resolved

**Comment by chalin** [comment]
@open-telemetry/java-approvers @open-telemetry/java-instrumentation-approvers PTAL

**Comment by zeitlinger** [comment]
@chalin can you check?

**Comment by chalin** [comment]
> @chalin can you check?

@zeitlinger - do you mean check if named regions would be a useful feature to apply in this PR? I won't have the time to do that. I suggest that we get this merged first.

We're still [waiting for the following approvals](https://github.com/open-telemetry/opentelemetry.io/pull/4642#issuecomment-2225841417) from Java SIG approvers:

- @open-telemetry/java-approvers
- @open-telemetry/java-instrumentation-approvers

**Comment by zeitlinger** [comment]
> @zeitlinger - do you mean check if named regions would be a useful feature to apply in this PR? I won't have the time to do that. I suggest that we get this merged first.

I actually like the fact that all files are complete and can be copied as is

**Comment by svrnm** [comment]
@zeitlinger can you check with Java SIG to have someone review and approve, then happy to merge this:-)

**Review by jack-berg — APPROVED** [review]
This is cool! I'm not familiar with all of the mechanics of how code snippets are referenced / injected, but the code in the java snippets looks good to me.

**Comment by svrnm** [comment]
> This is cool! I'm not familiar with all of the mechanics of how code snippets are referenced / injected, but the code in the java snippets looks good to me.

@chalin added some tooling for that from the dart project. We are still in a "public beta" phase of that tooling, but it looks promising, but it's also still a manual process to get things updated, etc.

Thanks for the review!

**Pull request merged**
